### PR TITLE
Document deployment profiles

### PR DESCRIPTION
**Pull request by jandubois**

### CODE REVIEW

**Comment by sunilarjun** [comment]
Smoke tested document addition for next successfully, merging in.

File: docs/getting-started/deployment.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+* Otherwise set any missing values to the builtin application defaults
+* Copy values from the "locked" profile over the current settings
+
+The user cannot modify any settings (via GUI or CLI) that have been locked by the profile.
+
+Rancher Desktop will refuse to load the application if a profile exists, but cannot be parsed correctly.
+
+Deployment profiles will not be modified or removed by Rancher Desktop. They will not be affected by a factory reset or uninstall.
+
+The structure of the profile data matches the application settins:
```
Comment (ericpromislow) [inline docs/getting-started/deployment.md:34]: Typo: `settings`

File: docs/getting-started/deployment.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Known Issues and Limitations
+
+* The `containerEngine.allowedImages` settings are currently the only ones that can be locked.
```
Comment (ericpromislow) [inline docs/getting-started/deployment.md:272]: This is not quite true.  `containerEngine.allowedImages.enabled` and `...patterns` are the only settings where if they're locked, it shows up in the UI. The UI disables them, shows a lock icon with a tooltip.

The UI gives the impression that other locked fields can be modified, but the user will get an error message if they try to apply the changes.

`rdctl set` will give an error if the command attempts to change a locked field's value (but it's not an error to try to change a locked field to the same value).

`rdctl start` and running the underlying app with a command-line option that would change a locked field's value should result in a failure to start the app.

File: docs/getting-started/deployment.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+* On macOS incorrectly formatted profiles are ignored instead of preventing the app from loading.
+* There is no way to set `diagnostics.showMuted` (and on Windows `WSL.integrations`) via deployment profile.
```
Comment (ericpromislow) [inline docs/getting-started/deployment.md:274]: The same holds for `extensions`, although I don't know if we plan to make that field directly writable, or only via some plugin mechanism.

File: docs/getting-started/deployment.md (hunk 1 above)
Comment (jandubois) [reply docs/getting-started/deployment.md:34]: @sunilarjun I'll create another PR based on Eric's feedback

File: docs/getting-started/deployment.md (hunk 3 above)
Comment (jandubois) [reply docs/getting-started/deployment.md:274]: There should be no `extensions` setting in `release-1.8`. It is part of RDX and should only exist on `main`.

File: docs/getting-started/deployment.md (hunk 1 above)
Comment (sunilarjun) [reply docs/getting-started/deployment.md:34]: Sounds good, thank you!

File: docs/getting-started/deployment.md (hunk 2 above)
Comment (jandubois) [reply docs/getting-started/deployment.md:272]: Ok, I will need to test this. This is not what we agreed on during the feature planning meeting, but we can leave it as-is and change the "Known issue".

Comment (ericpromislow) [reply docs/getting-started/deployment.md:272]: You're right -- I just looked at the code, and we only allow those `containerEngine.allowedImages` fields to be locked.

Comment (ericpromislow) [reply docs/getting-started/deployment.md:272]: I guess all my testing with the command-line and `rdctl set` fields were on those settings.

File: docs/getting-started/deployment.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+### Preferences values at startup
+
+Rancher Desktop settings are determined as follows:
+
+* Load "admin" deployment profile (both "defaults" and "locked")
+* If neither of them exist then load "user" deployment profile (again both "defaults" and "locked")
+* Load saved preferences from `settings.json` file
+* If there are no saved settings, use the "defaults" profile loaded earlier instead
+* If the "defaults" profile didn't exist either, show the first-run dialog
```
[the remark this answers is not in the document]
Comment (jandubois) [reply docs/getting-started/deployment.md:24]: This was already changed in the next PR to

> If the settings are still completely empty, show the first-run dialog

**Pull request merged**